### ISSUE TITLE
image not found in Mac OS 2.0.0 install from PyPi.

### ISSUE THREAD

**Issue by JamesB-1qbit** [pyscf/pyscf#1096]
In a fresh environment. I installed 
`python -m pip install pyscf`
The .whl that was downloaded was `pyscf-2.0.0-cp39-cp39-macosx_10_13_x86_64.whl`
It installed properly but when I tried to import pyscf.
```
Python 3.9.7 (default, Oct 13 2021, 06:45:31) 
[Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyscf
Traceback (most recent call last):
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/lib/misc.py", line 54, in load_library
    return numpy.ctypeslib.load_library(libname, _loaderpath)
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 449, in __getitem__
    return getattr(self, name)
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 444, in __getattr__
    dll = self._dlltype(name)
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/lib/libcgto.dylib, 6): Library not loaded: @rpath/libcint.4.dylib
  Referenced from: /Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/lib/libcgto.dylib
  Reason: image not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/__init__.py", line 114, in <module>
    from pyscf import gto
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/gto/__init__.py", line 19, in <module>
    from pyscf.gto import mole
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/gto/mole.py", line 46, in <module>
    from pyscf.gto import moleintor
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/gto/moleintor.py", line 29, in <module>
    libcgto = lib.load_library('libcgto')
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/lib/misc.py", line 62, in load_library
    return numpy.ctypeslib.load_library(libname, libpath)
  File "/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 449, in __getitem__
    return getattr(self, name)
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 444, in __getattr__
    dll = self._dlltype(name)
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/lib/libcgto.dylib, 6): Library not loaded: @rpath/libcint.4.dylib
  Referenced from: /Users/jamesbrown/Documents/Coding/environments/pyscftest2/lib/python3.9/site-packages/pyscf/lib/libcgto.dylib
  Reason: image not found
  ```
  
  When I go to `/lib/python3.9/site-packages/pyscf/lib/libcgto.dylib` and run
  ```
(pyscftest2) jamesbrown@MacBook Pro: jamesbrown lib % otool -l libcgto.dylib | grep libcint && otool -l libcgto.dylib | grep RPATH -A 2 
         name @rpath/libcint.4.dylib (offset 24)
          cmd LC_RPATH
      cmdsize 80
         path /Users/qssun/workspace/pyscf/pypi/pyscf-2.0/pyscf/lib/deps/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 80
         path /Users/qssun/workspace/pyscf/pypi/pyscf-2.0/pyscf/lib/deps/lib64 (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path /usr/local/lib64 (offset 12)
--
          cmd LC_RPATH
      cmdsize 72
         path /Users/qssun/workspace/pyscf/pypi/pyscf-2.0/pyscf/lib (offset 12)
```

The installation works when I build from setup.py after cloning the git repository.

**Comment by fishjojo** [comment]
Currently there is no wheel for python 3.9 on MacOS.

**Comment by JamesB-1qbit** [comment]
I'm not overly familiar with wheels but I believe there are wheels for Mac OS for python 3.6 to 3.9 on PyPi. https://pypi.org/project/pyscf/#files

In any case, `pip install pyscf` is not working for me for python 3.9 on Mac OS as it is trying to download and use `pyscf-2.0.0-cp39-cp39-macosx_10_13_x86_64.whl` and the import fails after installation.

**Comment by fishjojo** [comment]
I see. The dependency libraries are not correctly included in the wheels. @sunqm do you have any ideas?

**Comment by sunqm** [comment]
It's not about python3.9 on macos. Some problems exist in the packaging script for macos wheels. I removed the wheels from pypi. Now `pip install pyscf` should work on macos but may require a large amount of memory during compilation.

**Comment by JamesB-1qbit** [comment]
Thanks for the help so far! However, the installation now fails due to 
```
(pyscftest) jamesbrown@MacBook Pro: jamesbrown environments % python3 -m pip install pyscf --no-cache-dir
Collecting pyscf
  Downloading pyscf-2.0.1.tar.gz (7.7 MB)
     |████████████████████████████████| 7.7 MB 3.3 MB/s 
Collecting numpy!=1.16,!=1.17,>=1.13
  Downloading numpy-1.21.4-cp39-cp39-macosx_10_9_x86_64.whl (17.0 MB)
     |████████████████████████████████| 17.0 MB 12.4 MB/s 
Collecting scipy!=1.5.0,!=1.5.1
  Downloading scipy-1.7.1-cp39-cp39-macosx_10_9_x86_64.whl (32.8 MB)
     |████████████████████████████████| 32.8 MB 15.8 MB/s 
Collecting h5py>=2.7
  Downloading h5py-3.5.0-cp39-cp39-macosx_10_9_x86_64.whl (3.0 MB)
     |████████████████████████████████| 3.0 MB 10.1 MB/s 
Building wheels for collected packages: pyscf
  Building wheel for pyscf (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/jamesbrown/Documents/Coding/environments/pyscftest/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/setup.py'"'"'; __file__='"'"'/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-wheel-9wwmwshd
       cwd: /private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/
  Complete output (8 lines):
  running bdist_wheel
  running build
  running build_ext
  Configuring extensions
  cmake -S/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/pyscf/lib -Bbuild/temp.macosx-11-x86_64-3.9
  CMake Error: The source directory "/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/pyscf/lib" does not appear to contain CMakeLists.txt.
  Specify --help for usage, or press the help button on the CMake GUI.
  error: command '/usr/local/bin/cmake' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for pyscf
  Running setup.py clean for pyscf
Failed to build pyscf
Installing collected packages: numpy, scipy, h5py, pyscf
    Running setup.py install for pyscf ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/jamesbrown/Documents/Coding/environments/pyscftest/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/setup.py'"'"'; __file__='"'"'/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-record-3r7xj9il/install-record.txt --single-version-externally-managed --compile --install-headers /Users/jamesbrown/Documents/Coding/environments/pyscftest/include/site/python3.9/pyscf
         cwd: /private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/
    Complete output (8 lines):
    running install
    running build
    running build_ext
    Configuring extensions
    cmake -S/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/pyscf/lib -Bbuild/temp.macosx-11-x86_64-3.9
    CMake Error: The source directory "/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/pyscf/lib" does not appear to contain CMakeLists.txt.
    Specify --help for usage, or press the help button on the CMake GUI.
    error: command '/usr/local/bin/cmake' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/jamesbrown/Documents/Coding/environments/pyscftest/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/setup.py'"'"'; __file__='"'"'/private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-install-miycn_cf/pyscf_52c5b6aa669c46c297d4a07bdf88686a/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/dg/qjq47mdj0rg7svj15wzvmnhr0000gp/T/pip-record-3r7xj9il/install-record.txt --single-version-externally-managed --compile --install-headers /Users/jamesbrown/Documents/Coding/environments/pyscftest/include/site/python3.9/pyscf Check the logs for full command output.
  ```
  
  For your information
  ```
  (pyscftest) jamesbrown@MacBook Pro: jamesbrown environments % cmake --version
cmake version 3.21.4
```

**Comment by jamesETsmith** [comment]
@JamesB-1qbit, thanks for your patience with this. It looks like the `CMakeLists.txt` file is getting removed from the repo when the sources are packaged for pip, I tried "manually" downloading the tarred files from pip and there was no `CMakeLists.txt` file. Not sure why this is happening, in the meantime `pip install git+https://github.com/pyscf/pyscf` should work since the sources are pulled from Github.

**Comment by JoelHBierman** [comment]
@jamesETsmith Pip installing from GitHub does not appear to work either. It seems to get stuck building and installing libcx. You can work around this by running changing the compiler from the default clang to:

```
export CC=gcc-11
export CXX=g++-11
export CPPFLAGS="-I/opt/homebrew/include"
```

but then you get the same "image not found" error in the original post.

**Comment by kjelljorner** [comment]
@JoelHBierman I was able to install it with pip from GitHub with your exports. Not sure what would be different. I'm using HomeBrew-installed compilers etc under Rosetta 2 on a Mac M1.

**Comment by kjelljorner** [comment]
I think the problem here is that the libcint dylib is not installed for some reason. I compiled and installed it from the instructions here https://github.com/sunqm/libcint and after that I can install PySCF with pip from GitHub with @JoelHBierman's exports.

**Comment by JoelHBierman** [comment]
@kjelljorner I should caveat that this is trying to build for M1 natively. It definitely is possible to build libcint and pyscf for M1 using the "compile from source code" instructions at https://pyscf.org/install.html#compiling-from-source-code. I have done so (for 1.7) in the past, but installing using pip seems to present more obstacles. Of course, if you need to only run pyscf this is not an issue, but if pyscf is a dependency for other packages you want to use, then it is because generally those dependencies will end up trying to pip install pyscf anyways because it's not detected by the package manager.

Let me try following those instructions for building libcint and then pip installing pyscf separately.

**Comment by JoelHBierman** [comment]
@kjelljorner Is there a particular place libcint needs to be installed in order to be detected by pyscf in a fresh environment?

**Comment by kjelljorner** [comment]
@JoelHBierman I just did regular install which defaulted to `/usr/local/lib/`

**Comment by JoelHBierman** [comment]
Installing libcint and then pip installing from GitHub seems to do the trick on arm64. I don't get any import errors for just pure pyscf code. But now it seems I must open an issue for some packages which use pyscf as a dependency. If anything I dig up there points back to pyscf, I will open a separate issue if it seems appropriate.

**Comment by jamesETsmith** [comment]
Thanks for tracking this down @JoelHBierman !

**Comment by JoelHBierman** [comment]
I should caveat that my testing hasn't involved anything other than just importing pyscf, which is what will reproduce the problem of the original post. It could very well be that running actual calculations will cause more errors to pop up, but I cannot speak to how well that works currently.

**Comment by kjelljorner** [comment]
Not sure if it needs pointing out that this seems to be a problem that PySCF does not install libcint into the lib folder

**Comment by JoelHBierman** [comment]
It turns out that the error I uncovered at Qiskit/qiskit-nature#417 does not seem to have anything to do with the Qiskit package, but is in fact a pyscf issue. It's sufficiently similar to this issue that I am posting it here instead of creating a separate issue. I thought that testing the imports as `from pyscf import *` would act as a catch-all in case other libraries besides libcint were not being installed properly, but this turns out to not be the case. As you try to import specific things from pyscf, more errors related to other libraries not being loaded pop up. For example, running `from pyscf import dft` results in the following additional errors:

```
/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/dft/numint.py:31: UserWarning: XC functional libraries (libxc or XCfun) are not available.
  warnings.warn('XC functional libraries (libxc or XCfun) are not available.')
Traceback (most recent call last):
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/misc.py", line 54, in load_library
    return numpy.ctypeslib.load_library(libname, _loaderpath)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 457, in __getitem__
    return getattr(self, name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 452, in __getattr__
    dll = self._dlltype(name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 382, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib, 6): Library not loaded: @rpath/libxc.9.dylib
  Referenced from: /Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib
  Reason: image not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/dft/numint.py", line 25, in <module>
    from pyscf.dft import libxc
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/dft/libxc.py", line 35, in <module>
    _itrf = lib.load_library('libxc_itrf')
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/misc.py", line 62, in load_library
    return numpy.ctypeslib.load_library(libname, libpath)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 457, in __getitem__
    return getattr(self, name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 452, in __getattr__
    dll = self._dlltype(name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 382, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib, 6): Library not loaded: @rpath/libxc.9.dylib
  Referenced from: /Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib
  Reason: image not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/misc.py", line 54, in load_library
    return numpy.ctypeslib.load_library(libname, _loaderpath)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 457, in __getitem__
    return getattr(self, name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 452, in __getattr__
    dll = self._dlltype(name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 382, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxcfun_itrf.dylib, 6): Library not loaded: @rpath/libxcfun.2.dylib
  Referenced from: /Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxcfun_itrf.dylib
  Reason: image not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/joelbierman/Desktop/Misc./Old Code/Qiskit_Projects_Migration/CI_tests.py", line 3, in <module>
    from pyscf import ci, gto, scf, dft
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/dft/__init__.py", line 49, in <module>
    from pyscf.dft import numint
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/dft/numint.py", line 28, in <module>
    from pyscf.dft import xcfun
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/dft/xcfun.py", line 32, in <module>
    _itrf = lib.load_library('libxcfun_itrf')
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/misc.py", line 62, in load_library
    return numpy.ctypeslib.load_library(libname, libpath)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 457, in __getitem__
    return getattr(self, name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 452, in __getattr__
    dll = self._dlltype(name)
  File "/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/ctypes/__init__.py", line 382, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxcfun_itrf.dylib, 6): Library not loaded: @rpath/libxcfun.2.dylib
  Referenced from: /Users/joelbierman/miniforge3/envs/Qiskit-0280-ARM64-pyscf2/lib/python3.9/site-packages/pyscf/lib/libxcfun_itrf.dylib
  Reason: image not found

```

In addition to libcint, it seems that libxc and libxcfun are also not being installed. Is there a full list of libraries that should be checked for this same error and recommended methods or instructions for installing them?

**Comment by sunqm** [comment]
Pip wheels for macos were added. Running `pip install pyscf==2.0.1` should be enough now

**Comment by sunqm** [comment]
> Installing libcint and then pip installing from GitHub seems to do the trick on arm64. (I imagine that it would also work for x86 and rosetta) I don't get any import errors for just pure pyscf code. But now it seems I must open an issue for some packages which use pyscf as a dependency. If anything I dig up there points back to pyscf, I will open a separate issue if it seems appropriate. Thank you for your help!

In next release, the source code and makefiles will be included in the sdist file on pypi.

**Comment by JamesB-1qbit** [comment]
The new wheel worked for me! I assume this issue can be closed now?

**Comment by JoelHBierman** [comment]
I can confirm that pip installing from the master GitHub branch also works on arm64 Macs, with the caveat that you can't use the default clang compiler. You have to use the exports:

```
export CC=gcc-11
export CXX=g++-11
export CPPFLAGS="-I/opt/homebrew/include"
```

A couple things are not 100% clear to me that I should note: 1. I don't know if these exports are necessary for x86. and 2. I don't know if the third export line is necessary. This is just something that was suggested to me on a separate issue and it seems to work.

Given that there are no published macosx arm64 wheels on pypi, would it be possible to add some kind of note on the installation page https://pyscf.org/install.html (or somewhere else) regarding the couple of extra steps that are needed to build for this architecture? (i.e. use a compiler that Isn't clang and *possibly* have to install from GitHub, although it's not clear to me if this will be necessary after the next version release or if it's specific to 2.0.1.) Ultimately installing pyscf for both architectures is now straightforward, but making some kind of note regarding arm64 builds might save some users (particularly beginners) time and energy as well as sparing them some frustration.

**Comment by JamesB-1qbit** [comment]
Another image not found error on MacOS. pyscf==2.0.1
```
from pyscf.dft import libxc
Traceback (most recent call last):
  File "/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/lib/misc.py", line 54, in load_library
    return numpy.ctypeslib.load_library(libname, _loaderpath)
  File "/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 449, in __getitem__
    return getattr(self, name)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 444, in __getattr__
    dll = self._dlltype(name)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib, 6): Library not loaded: @rpath/libxc.9.dylib
  Referenced from: /Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib
  Reason: image not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/dft/libxc.py", line 35, in <module>
    _itrf = lib.load_library('libxc_itrf')
  File "/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/lib/misc.py", line 62, in load_library
    return numpy.ctypeslib.load_library(libname, libpath)
  File "/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/numpy/ctypeslib.py", line 149, in load_library
    return ctypes.cdll[libpath]
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 449, in __getitem__
    return getattr(self, name)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 444, in __getattr__
    dll = self._dlltype(name)
  File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib, 6): Library not loaded: @rpath/libxc.9.dylib
  Referenced from: /Users/jamesbrown/Documents/Coding/environments/qsdkmain/lib/python3.9/site-packages/pyscf/lib/libxc_itrf.dylib
  Reason: image not found
```

**Comment by MarcoBarroca** [comment]
Have there been any updates to this now that v2.1 released?

**Comment by sunqm** [comment]
This issue should have been fixed in pyscf-2.2

**Comment by kjelljorner** [comment]
I think a follow-up issue is that the wheels is [listed as universal](https://pypi.org/project/pyscf/#files) for Python 3.11, while this doesn't work.
```
OSError: dlopen(/Users/Kjell/mambaforge/envs/pyscf/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib, 0x0006): tried: '/Users/Kjell/mambaforge/envs/pyscf/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/Kjell/mambaforge/envs/pyscf/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (no such file), '/Users/Kjell/mambaforge/envs/pyscf/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```
